packages/flutter_tools/lib/src/base/net.dart

```diff
@@ -57,7 +57,8 @@ Future<List<int>> _attempt(Uri url) async {
     final BytesBuilder responseBody = new BytesBuilder(copy: false);
     await response.forEach(responseBody.add);
     return responseBody.takeBytes();
-  } on IOException catch (error) {
+  } catch (error) {
```

Comment:
I would rather we just listed all the known possible exceptions explicitly. Several times so far we've had cases where we have a bug but we swallow all the exceptions somewhere and so miss the bug, because we end up swallowing an ArgumentError or AssertionError or whatnot.

Reply:
+1